packages/api/src/routes/nfts-upload.js

```diff
@@ -66,8 +71,9 @@ export async function upload(event, ctx) {
   }
 
   let pin = await pins.get(nft.cid)
-  if (!pin || pin.status !== 'pinned') {
-    pin = { cid: nft.cid, status: 'pinned', size: nftSize, created }
+  if (!pin) {
+    const status = cluster.toPSAStatus(await cluster.status(nft.cid))
```

Comment:
at some point, we need to make the status as pinned

Reply:
There is a cron job! https://github.com/ipfs-shipyard/nft.storage/blob/main/packages/cron/src/jobs/pins.js

If you call `pins.set` with a pin that is not "pinned", it'll actually add an entry to the `FOLLOWUPS` KV. The cron job iterates over everything and checks the status with IPFS Cluster. If the status changed it saves the new value, and if it becomes pinned it asks cluster for it's DAG size and saves that too.